solr/solrj/src/java/org/apache/solr/common/util/XML.java

```diff
@@ -208,7 +179,13 @@ public static void escape(String str, Writer out, String[] escapes) throws IOExc
     }
   }
 
+  @FunctionalInterface
   public interface Writable {
-    void write(Writer w) throws IOException ;
+    void write(Writer w) throws IOException;
+  }
+
+  @FunctionalInterface
+  public interface Escapable<T, U> {
```

Comment:
Perhaps since this is a utility some user will have a use case to escape their own attributes, like we did with the data in the child document. So I added this.

Reply:
It's somewhat unusual to have pre-escaped content, I think.  Case in point: the "unescape" method, judging from "find usages" in my IDE, is only used in tests.  Furthermore, I think it's not a good idea to introduce an interface or class to serve a use case that does not yet exist, nor do we particularly think it's going to happen.